packages/components/src/templates/next/components/internal/Search.tsx

```diff
@@ -64,32 +51,67 @@ export function SearchField({
   placeholder,
   ...props
 }: SearchFieldProps) {
+  const inputRef = useRef<HTMLInputElement>(null)
+  const [isFocusWithin, setIsFocusWithin] = useState(false)
+  const { inputProps } = useTextField(props, inputRef)
+  const { focusProps } = useFocusRing()
+
+  const isDisabled = props.isDisabled ?? false
+  const isInvalid = props.isInvalid ?? false
+  const value = typeof inputProps.value === "string" ? inputProps.value : ""
+  const isEmpty = !value || value.length === 0
+
+  const handleClear = () => {
+    if (props.onChange) {
+      props.onChange("")
+    }
+    inputRef.current?.focus()
+  }
+
   return (
-    <AriaSearchField
-      {...props}
-      className={composeTailwindRenderProps(
-        classNames?.container,
-        "group flex flex-col gap-2",
-      )}
-    >
-      <FieldGroup className={classNames?.fieldgroup}>
+    <div className={twMerge("flex flex-col gap-2", classNames?.container)}>
+      <div
+        className={twMerge(
+          fieldGroupStyles({
+            isFocusWithin,
+            isInvalid,
+            isDisabled,
+          }),
+          classNames?.fieldgroup,
+        )}
+        onFocus={() => setIsFocusWithin(true)}
+        onBlur={(e) => {
+          if (!e.currentTarget.contains(e.relatedTarget as Node)) {
+            setIsFocusWithin(false)
+          }
+        }}
+      >
         <BiSearch
           aria-hidden
-          className="h-5 w-5 text-base-content-medium forced-colors:text-[ButtonText] forced-colors:group-disabled:text-[GrayText]"
+          className={
+            isDisabled
+              ? "h-5 w-5 text-base-content-medium forced-colors:text-[GrayText]"
+              : "h-5 w-5 text-base-content-medium forced-colors:text-[ButtonText]"
+          }
         />
-        <Input
-          className={composeTailwindRenderProps(
-            classNames?.input,
+        <input
+          {...mergeProps(inputProps, focusProps)}
+          ref={inputRef}
+          type="search"
+          className={twMerge(
+            inputStyles(),
             "bg-transparent [&::-webkit-search-cancel-button]:hidden",
+            classNames?.input,
           )}
           placeholder={placeholder}
```

Comment:
[comment] from the react aria repo, we should apply an empty space string for safari lolol: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/s2/src/TextField.tsx#L191